src/charm.py

```diff
@@ -80,11 +85,45 @@ def _on_config_changed(self, _):
         else:
             self.unit.status = WaitingStatus("Waiting for Pebble in workload container")
 
+    def _patch_auth_resources(self):
+        """Returns True if existing auth resources for training-operator are patched correctly."""
+        self.unit.status = MaintenanceStatus("Patching auth configuration")
+        client = Client()
+        context = {"namespace": self._namespace}
+
+        with open(os.path.join(self._src_dir, "auth_manifests.yaml")) as f:
```

Comment:
```suggestion
        with open(Path(self._src_dir) / "auth_manifests.yaml") as f:
```

Just a nit, but I prefer using `pathlib` where possible :)